File: jung-visualization/src/main/java/edu/uci/ics/jung/visualization/spatial/rtree/LeafSplitter.java
package edu.uci.ics.jung.visualization.spatial.rtree;

import java.awt.geom.Rectangle2D;
import java.util.Collection;
import java.util.Map;

/**
 * Interface for splitting LeafNodes containing Map.Entries as children
 *
 * @param <T> the type of the elements
 * @author Tom Nelson
 */
public interface LeafSplitter<T> {

  Pair<LeafNode<T>> split(
      Collection<Map.Entry<T, Rectangle2D>> entries, Map.Entry<T, Rectangle2D> newEntry);
}

File: jung-visualization/src/main/kotlin/edu/uci/ics/jung/visualization/spatial/rtree/LeafSplitter.kt
package edu.uci.ics.jung.visualization.spatial.rtree

import java.awt.geom.Rectangle2D

/**
 * Interface for splitting LeafNodes containing Map.Entries as children
 *
 * @param T the type of the elements
 * @author Tom Nelson
 */
interface LeafSplitter<T> {
  fun split(
    entries: Collection<Map.Entry<T, Rectangle2D>>,
    newEntry: Map.Entry<T, Rectangle2D>
  ): Pair<LeafNode<T>>
}
